src/json/builder.cr

```diff
@@ -94,35 +94,55 @@ class JSON::Builder
   # This method can also be used to write the name of an object field.
   def string(value)
     string = value.to_s
+
     scalar(string: true) do
       io << '"'
-      string.each_char do |char|
-        case char
+
+      pos = 0
+      reader = Char::Reader.new(string)
+
+      while reader.has_next?
```

Comment:
You could use `while char = reader.next_char` (obviously modified to include the first char), eliminating an additional method call.

Reply:
no. next_char raises on end

Reply:
This can easily be handled with `when Char::ZERO; break` and the reader will never get to a point where it raises. This solution should be faster than calling `has_next?` *and* `current_char` on each iteration.

Reply:
string can have zero byte in it. so condition not always work.

Reply:
True, that's a valid character in a JS string.
Still, you could check `has_next?` only if `char == Char::ZERO` instead of every iteration.

Reply:
search `while reader.has_next?` in std. 7 occurrences. nothing wrong with this code. plus i dont think its slower than your approach. this is idiomatic code.